pkgs/applications/audio/spotify-player/default.nix

```diff
@@ -4,12 +4,30 @@
 , pkg-config
 , openssl
 , cmake
+# deps for audio backends
 , alsa-lib
+, libpulseaudio
+, portaudio
+, libjack2
+, SDL2
+, gst_all_1
 , dbus
 , fontconfig
 , libsixel
+
+# build options
+, withStreaming ? true
+, withDaemon ? true
+, withAudioBackend ? "rodio" # alsa, pulseaudio, rodio, portaudio, jackaudio, rodiojack, sdl
+, withMediaControl ? true
+, withLyrics ? true
+, withImage ? true
+, withNotify ? true
+, withSixel ? true
 }:
 
+assert lib.assertOneOf "withAudioBackend" withAudioBackend [ "" "alsa" "pulseaudio" "rodio" "portaudio" "jackaudio" "rodiojack" "sdl" "gstreamer" ];
```

Comment:
Should it be built without any audio backends `""`?

Reply:
there is no reason why not. One might want to just use it as a client for spotify connect. It builds and runs afaict